src/nupic/algorithms/Connections.hpp

```diff
@@ -41,15 +41,15 @@ namespace nupic
 
     namespace connections
     {
-      typedef UInt32 CellIdx;
-      typedef unsigned char SegmentIdx;
-      typedef unsigned char SynapseIdx;
+      typedef UInt CellIdx;
```

Comment:
Okay, the CellIdx change makes sense. Regarding the SegmentIdx change, do we need to create an invalid Segment? Could we just use pointers instead of references in bestMatchingCell / bestMatchingSegment and return null?